packages/react-ui-core/src/Modal/ClosableModal.js

```diff
@@ -0,0 +1,72 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import themed from 'react-themed'
+import classnames from 'classnames'
+import autobind from 'autobind-decorator'
+
+import { Modal } from '../Modal'
+
+@themed(/^ClosableModal/)
+export default class ClosableModal extends Component {
+  static propTypes = {
+    className: PropTypes.string,
+    theme: PropTypes.object,
+    isOpen: PropTypes.bool,
+    onClose: PropTypes.func,
+    children: PropTypes.any,
+  }
+
+  static defaultProps = {
+    theme: {},
+    isOpen: false,
+  }
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      isOpen: props.isOpen,
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.isOpen !== this.props.isOpen) {
+      this.setState({ isOpen: nextProps.isOpen })
+    }
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
+    return nextProps.isOpen !== this.props.isOpen ||
```

Comment:
So basically anything here will return truthy?

Reply:
No, if it was set to true and it state true, it would return false...

Reply:
So you're updating when it's either open or closed? I guess I'm not seeing why you just wouldn't make this a PureComponent instead.  You're pretty much doing the same thing.

Reply:
Fair enough. I think making this a pure component could be better.